gnat2goto/driver/tree_walk.adb

```diff
@@ -2430,6 +2434,78 @@ package body Tree_Walk is
    ---------------------------
 
    procedure Do_Object_Declaration (N : Node_Id; Block : Irep) is
+      Obj_Id : constant Symbol_Id :=
+        Intern (Unique_Name (Defining_Identifier (N)));
+   begin
+      --  First check for object declarations which are not constants
+      if not Constant_Present (N) then
+         --  Not any sort of constant.
+         --  Process non-constant object_declaration.
+         Do_Object_Declaration_Full_Declaration (N, Block);
+      elsif --  Check that this isn't a completion of a deferred constant.
+         not Global_Symbol_Table.Contains (Obj_Id)
+      then
+         --  The declaration is of constant which may be deferred.
```

Comment:
It is about constant variables. It seems that we need to handle them differently since one can declare variables of incomplete types as long as they are constants.

Reply:
You can defer the declaration of a constant of a private type.  This means that there are two declarations of the constant and we need to tie them together.  In the code, on encountering the deferred constant declaration, its later full declaration is obtained from the tree.  When the later full declaration is encountered during the tree parse we do not want to process it again.